crates/dbsp/src/operator/filter_map.rs

```diff
@@ -211,7 +214,141 @@ where
     }
 }
 
-impl<C, K, V, R> FilterMap<C> for Stream<C, OrdIndexedZSet<K, V, R>>
+impl<C, K, R> FilterMap<C> for Stream<C, FileZSet<K, R>>
+where
+    C: Circuit,
+    K: DBData,
+    R: DBWeight,
+{
+    type Item = K;
+    type ItemRef<'a> = &'a K;
+    type R = R;
+
+    fn filter<F>(&self, filter_func: F) -> Self
```

Comment:
I assume this code is identical with the one for non-persistent streams below (although I do see at least one difference). 
If it is, can that be avoided?
If it's not, whats the difference?

Reply:
It's identical. I couldn't figure out a way to avoid it except by writing a macro, because otherwise Rust tells me that I'm doing trait specialization, which it doesn't support.

I could write a macro if you want.

Reply:
forget about the macro, just wanted to know.
If this is all the functions you need, it's ok, but if you need more it may become a problem.
Maybe you can add a comment explaining this.